src/main/src/test/java/org/geoserver/catalog/MockRetypeFeatureTypeCallback.java

```diff
@@ -0,0 +1,71 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import org.geoserver.catalog.retype.MockRetypedSource;
+import org.geotools.data.FeatureSource;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.referencing.CRS;
+import org.locationtech.jts.geom.Point;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.FeatureType;
+import org.opengis.feature.type.PropertyDescriptor;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * This is a mock implementation
+ *
+ * @author ImranR
+ */

```

Comment:
Noisy JavaDoc, better no doc that noise, you can configure your IDE to not create this templates. Since there is not final dot in your sentence I'm not even sure you finished it ...

Reply:
will change